### PR TITLE
Use the dataType if the baseType is not set

### PR DESCRIPTION
**Pull request by JustinBusschau**
A fix for #4873 

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@taxpon @frol @mbohlool @cbornet @kenjones-cisco @tomplus @Jyhess @slash-arun @spacether

### CODE REVIEW

**Comment by JustinBusschau** [comment]
I have left a single space before the `dataType`

```
{{paramName}} = {{#baseType}}{{baseType}}{{/baseType}}{{^baseType}}{{#dataType}} {{dataType}}{{/dataType}}{{/baseType}}.from_dict(connexion.request.get_json())  # noqa: E501
```

If I remove that space, the `dataType` is not rendered and we end up with the same result as raised in #4873 (the invalid `.from_dict).

Any suggestions on how to correct this are welcome.

**Comment by spacether** [comment]
Thank you for this PR! Can you please add a sample model/endpoint whixh shows this working in the flask samples folder?
- the flask sample is here https://github.com/OpenAPITools/openapi-generator/tree/master/samples/server/petstore/python-flask 
- a test should be added here

**Comment by JustinBusschau** [comment]
> * a test should be added here https://github.com/OpenAPITools/openapi-generator/tree/master/samples/server/petstore/python-flask/openapi_server/test

@spacether 
The specific case that this fix addresses is not covered by the current definition in `petstore.yaml`. This only occurs when you send an object or an enum as a query parameter. Neither of these cases are defined in the `petstore.yaml` - how should I proceed?

**Comment by spacether** [comment]
@JustinBusschau can you make a new version of petstore.yaml which includes your use case?
Then you can regenerate samples and add python tests of it.

**Comment by spacether** [comment]
@JustinBusschau it would be easier if you used the existing sh sample generation scripts. That way,
all you have to do is 
- save off a new version of the spec
- edit the sh file to point to the new spec
- add your new sample models
- run the sh file
- add the tests to the new sample files

Can you make those changes?
Doing it this way will minimize the number of files that need to be reviewed

**Comment by JustinBusschau** [comment]
The difficulty I had doing it that way is that the changes I introduced to the spec are only supported by `openapi: 3.0.0` and not by `swagger: 2.0`

Editing the existing `sh` file to use my altered spec looks like moving away from supporting `swagger` specs with this generator - which was not the intention of this change. I was hoping to have this be a change that supports a feature of `openapi` without affecting those who still need to use `swagger` specs with this generator - which I need to do with some of my older-but-still-used projects. I want to leave the older `swagger`-style test (without the objects in request parameters) and add a new test that checks specifically for the new `openapi`-only feature.

**Comment by spacether** [comment]
> The difficulty I had doing it that way is that the changes I introduced to the spec are only supported by `openapi: 3.0.0` and not by `swagger: 2.0`
> 
> Editing the existing `sh` file to use my altered spec looks like moving away from supporting `swagger` specs with this generator - which was not the intention of this change. I was hoping to have this be a change that supports a feature of `openapi` without affecting those who still need to use `swagger` specs with this generator - which I need to do with some of my older-but-still-used projects. I want to leave the older `swagger`-style test (without the objects in request parameters) and add a new test that checks specifically for the new `openapi`-only feature.

So this generator already has an openapi (v3) spec sample per:
- https://github.com/OpenAPITools/openapi-generator/blob/master/bin/openapi3/python-flask-petstore-python2.sh
- https://github.com/OpenAPITools/openapi-generator/blob/master/bin/openapi3/python-flask-petstore.sh

Both of those use petstore.yaml at `modules/openapi-generator/src/test/resources/3_0/petstore.yaml`
Editing those existing two sh files is not moving away from supporting swagger specs because that definition is v3 definition.
For reference swagger(v2) generation is done in:
- https://github.com/OpenAPITools/openapi-generator/blob/master/bin/python-server-flask-petstore-python2.sh
- https://github.com/OpenAPITools/openapi-generator/blob/master/bin/python-server-flask-petstore.sh
Which uses a different yaml spec to generate samples and should address your concern about supporting swagger users.

My concerns with doing it the way that you have done are:
- it is confusing to users which sample folder to look at for a v3 spec generated sample
  - samples/server/petstore/ (historically we have stored v2 samples here not v3)
  - samples/openapi3/server/petstore/python-flask
  - samples/openapi3/server/petstore/python-flask-python2
- If all other generators follow this precendent, then we could have 3+ samples folders per generator per spec which will lead to:
  - many more samples
  - increased CI runtime
  - user confusion about where to go to look at a sample of the generator output

Why not just save off your new spec at:
`https://github.com/OpenAPITools/openapi-generator/tree/master/modules/openapi-generator/src/test/resources/3_0/flask/petsote.yaml`?

**Comment by JustinBusschau** [comment]
Of course - I was missing the `openapi3` folder in `/bin` and consequently using the wrong sample generation scripts. Thanks for pointing this out @spacether :)

File: modules/openapi-generator/src/main/resources/python-flask/requirements.mustache [hunk 1]
```diff
@@ -1,4 +1,4 @@
-connexion >= 2.5.0; python_version>="3.6"
+connexion >= 2.6.0; python_version>="3.6"
```
Comment (spacether) [inline modules/openapi-generator/src/main/resources/python-flask/requirements.mustache]: Why did you change this version?

Comment (JustinBusschau) [reply modules/openapi-generator/src/main/resources/python-flask/requirements.mustache]: When using 2.5.0 I was getting an error about not being able to import `FileStorage` from `werkzeug `. The only way I could find to reliably fix this was to upgrade connexion as per [this answer](https://github.com/zalando/connexion/issues/1149#issuecomment-583848140) to [this issue](https://github.com/zalando/connexion/issues/1149)

**Comment by JustinBusschau** [comment]
Any advances on this @spacether ?

If there's a strong argument to be made for keeping the older version of connexion for python versions >= 3.6 I'm happy to roll back that change - addressing the `FileStorage` issue for python versions < 3.6 as you suggest in https://github.com/OpenAPITools/openapi-generator/issues/5235 is obviously not within the remit of this change

**Comment by wing328** [comment]
@JustinBusschau thanks for the PR, which has been included in the 4.3.0 release: https://twitter.com/oas_generator/status/1243455743937789952
